### PR TITLE
Save log in file

### PR DESCRIPTION
**Pull request by miiu96**
Added an extra flag that will specify to save logs in a file.
Also now proxy can start with different log levels ( --log-level *:INFO, DEBUG or TRACE)

### CODE REVIEW

File: cmd/proxy/config/config.toml [hunk 1]
```diff
@@ -42,16 +42,16 @@
 # shard id 4294967295
 [[Observers]]
    ShardId = 0
-   Address = "http://127.0.0.1:8080"
+   Address = "http://127.0.0.1:8081"
```
Comment (bogdan-rosianu) [inline cmd/proxy/config/config.toml:45]: rollback these changes

File: cmd/proxy/main.go [hunk 2]
```diff
@@ -138,6 +170,31 @@ func main() {
 }
 
 func startProxy(ctx *cli.Context) error {
+	logLevelFlagValue := ctx.GlobalString(logLevel.Name)
```
Comment (bogdan-rosianu) [inline cmd/proxy/main.go:173]: you can extract the logger handling code in a new function

File: cmd/proxy/main.go [hunk 3]
```diff
@@ -138,6 +170,31 @@ func main() {
...
+	}
+
+	if !check.IfNil(fileLogging) {
+		err = fileLogging.ChangeFileLifeSpan(time.Second * time.Duration(logFileLifeSpanInSec))
+		if err != nil {
+			return err
+		}
+	}
+
+	log.Trace("message", "bla", "bla")
```
Comment (bogdan-rosianu) [inline cmd/proxy/main.go:196]: remove this

File: cmd/proxy/main.go [hunk 4]
```diff
@@ -94,11 +103,31 @@ VERSION:
...
+			" the logs for all packages will have the INFO level, excepting the api package which will receive a DEBUG" +
+			" log level.",
+		Value: "*:" + logger.LogInfo.String(),
+	}
+	//logFile is used when the log output needs to be logged in a file
+	logSaveFile = cli.BoolFlag{
+		Name:  "log-save",
+		Usage: "Boolean option for enabling log saving. If set, it will automatically save all the logs into a file.",
+	}
+	// workingDirectory defines a flag for the path for the working directory.
```
Comment (bogdan-rosianu) [inline cmd/proxy/main.go:120]: you can drop this parameter and keep only the logs directory
anyway, if you keep it, at least change the `Usage`

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: go.mod [hunk 5]
```diff
@@ -3,9 +3,9 @@ module github.com/ElrondNetwork/elrond-proxy-go
 go 1.12
 
 require (
-	github.com/ElrondNetwork/elrond-go v1.1.1-0.20200914111912-d1792d662b8f
+	github.com/ElrondNetwork/elrond-go v1.1.2-0.20201016080150-915a6ef433de
```
Comment (andreibancioiu) [inline go.mod:6]: Not sure, but - can we reference `v1.1.3` already perhaps?

**Review by andreibancioiu — COMMENTED** [review]
Along awaited feature :100: :partying_face: 

Please resolve @bogdan-rosianu's comments.

File: cmd/proxy/config/config.toml (hunk 1 above)
Comment (miiu96) [reply cmd/proxy/config/config.toml:45]: if you start proxy as is with that configuration you will have a loop. proxy starts at `localhost:8080` and you have and observer with `localhost:8080` and you will have a infinite loop of requests.

File: cmd/proxy/main.go (hunk 2 above)
Comment (miiu96) [reply cmd/proxy/main.go:173]: done.

File: cmd/proxy/main.go (hunk 3 above)
Comment (miiu96) [reply cmd/proxy/main.go:196]: done.

File: cmd/proxy/main.go (hunk 4 above)
Comment (miiu96) [reply cmd/proxy/main.go:120]: done.

File: cmd/proxy/config/config.toml (hunk 1 above)
Comment (bogdan-rosianu) [reply cmd/proxy/config/config.toml:45]: ok. good catch.

**Review by bogdan-rosianu — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**